crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -432,20 +432,34 @@ impl EventHandler {
                     "Intercepted HTLC"
                 );
 
+                let fake_channel_payments = self.fake_channel_payments.lock().unwrap();
+
+                let target_node_id = match fake_channel_payments.get(&requested_next_hop_scid) {
+                    None => {
+                        tracing::warn!(fake_scid = requested_next_hop_scid, "Could not forward the intercepted HTLC because we didn't have a node registered with said fake scid");
+
+                        if let Err(err) = self.channel_manager.fail_intercepted_htlc(intercept_id) {
+                            tracing::error!("Could not fail intercepted htlc {err:?}")
+                        }
+
+                        return Ok(());
+                    }
+                    Some(target_node_id) => target_node_id,
+                };
+
                 // if we have already a channel with them, we try to forward the payment.
-                // TODO: here we would need to increase the channel size if the channel is too small
-                if let Some(channel) =
-                    self.channel_manager
-                        .list_channels()
-                        .iter()
-                        .find(|channel_details| {
-                            if let Some(scid) = channel_details.short_channel_id {
-                                scid == requested_next_hop_scid
-                            } else {
-                                false
-                            }
-                        })
+                if let Some(channel) = self
+                    .channel_manager
+                    .list_channels()
+                    .iter()
+                    .find(|channel_details| channel_details.counterparty.node_id == *target_node_id)
```

Comment:
🔧 It would help to leave a comment about why this should work e.g.
```suggestion
                    // The coordinator can only have one channel with each app. Hence, if we find a channel
                    // with the target of the intercepted HTLC, we know that it is the only channel between
                    // coordinator and target app and we can forward the intercepted HTLC through it.
                    .find(|channel_details| channel_details.counterparty.node_id == *target_node_id)
```